src/javascripts/components/search.tracking.js

```diff
@@ -0,0 +1,119 @@
+import 'govuk-frontend/vendor/polyfills/Element/prototype/classList'
+import common from 'govuk-frontend/common'
+var nodeListForEach = common.nodeListForEach
+
+function addToDataLayer (payload) {
+  window.dataLayer = window.dataLayer || []
+  window.dataLayer.push(payload)
+}
+
+function stripPossiblePII (string) {
+    // Try to detect emails and redact it.
+  string = string.replace(/\S*@\S*\s?/g, '[blocked]')
+    // If someone has typed in a number it's likely not related so redact it
+  string = string.replace(/0|1|2|3|4|5|6|7|8|9/g, '[blocked]')
```

Comment:
Minor, but if we replaced `/[0-9]+/g` I think we'd end up replacing any groups of numbers with [blocked], rather than each number individually?

e.g. `Hey 12345 Foo6789` would become `Hey [blocked] Foo[blocked]` rather than `Hey [blocked][blocked][blocked][blocked][blocked] Foo[blocked][blocked][blocked][blocked]`

Which seems more readable / manageable?

Reply:
This regex is based from what Daisy originally wrote that I've double checked with her but will see what she thinks.

Reply:
:+1: